opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/WebOpacAt.java

```diff
@@ -0,0 +1,290 @@
+            if (desc != null && !desc.trim().isEmpty()) {
+                result.addDetail(new Detail(desc, content));
+            } else if (!result.getDetails().isEmpty()) {
+                final Detail lastDetail = result.getDetails().get(result.getDetails().size() - 1);
+                lastDetail.setHtml(true);
+                lastDetail.setContent(lastDetail.getContent() + "<br>" + content);
+            }
+        }
+        return result;
+    }
+
+    private String getCellContent(Element detailTable, String pattern) {
+        final Element first = detailTable.select("td.label:matchesOwn(" + pattern + ")").first();
+        return first == null ? null : first.nextElementSibling().text();
+    }
+
+    private static String getCover(Element doc) {
+        return doc.select(".coverimage img").first().attr("src").replaceFirst("&width=\\d+", "");
+    }
+
+    @Override
+    public SearchRequestResult filterResults(Filter filter, Filter.Option option)
+            throws IOException, OpacErrorException {
+        return null;
+    }
+
+    @Override
+    public DetailledItem getResult(int position) throws IOException, OpacErrorException {
+        return null;
+    }
+
+    @Override
+    public List<SearchField> getSearchFields()
+            throws IOException, OpacErrorException, JSONException {
+        start();
+        final List<SearchField> fields = new ArrayList<>();
+        getSimpleSearchField(fields);
+        getAdvancedSearchFields(fields);
+        return fields;
+    }
+
+    protected void getSimpleSearchField(List<SearchField> fields)
+            throws IOException, JSONException {
+        final String html = httpGet(opac_url + "/search?mode=s", getDefaultEncoding());
+        final Document doc = Jsoup.parse(html);
+        final Element simple = doc.select(".simple_search").first();
+        final TextSearchField field = new TextSearchField();
+        field.setFreeSearch(true);
+        field.setDisplayName(simple.select("h4").first().text());
+        field.setId(simple.select("#keyboard").first().attr("name"));
+        field.setHint("");
+        field.setData(new JSONObject());
+        field.getData().put("meaning", field.getId());
+        fields.add(field);
+    }
+
+    protected void getAdvancedSearchFields(List<SearchField> fields)
+            throws IOException, JSONException {
+        final String html = httpGet(opac_url + "/search?mode=a", getDefaultEncoding());
+        final Document doc = Jsoup.parse(html);
+        final Elements options = doc.select("select#adv_search_crit_0").first().select("option");
+        for (final Element option : options) {
+            final TextSearchField field = new TextSearchField();
+            field.setDisplayName(option.text());
+            field.setId(option.val());
+            field.setHint("");
+            field.setData(new JSONObject());
+            field.getData().put("meaning", field.getId());
+            fields.add(field);
+        }
```

Comment:
So the `SearchField`s are not only used to specify a search criterion/filter, but also for sorting? Would that just be an indistinctive `SearchField` that can be added to the query?

"Antolin" should be rather easy to add, although I'm not overly interested in that criterion :smirk:. "Suche in/Partnerbibliothek" adds a state to performing queries, since that is set via a different HTTP request.

Another enhancement would be to obtain all possible values for certain fields (such as "Systematik|Classification", via http://www.aktirol.web-opac.at/search/adv_ac?crit=og) and add a `DropdownSearchField` instead …